src/main.rs

```diff
@@ -1,35 +1,232 @@
+use std::{error::Error, fmt::Display};
+
 use smbioslib::*;
 
+#[derive(Debug)]
+enum BiosParseError {
```

Comment:
Should be this enum leave somewhere else than`main.rs`?

Reply:
I'm thinking maybe we want to start a dmidecode-rs github project so it is not seen as part of a library but something that stands on its own?  It will likely draw attention of its own since it is an established keyword and product.

Reply:
Just now created the repo for it (empty at the moment).  Will move/copy code over from here.